### PR TITLE
Support ApplyParentClosePolicy Cross Cluster Tasks

### PR DESCRIPTION
**Pull request by demirkayaender**
**What changed?**
With these changes we can persist cross cluster requests to apply parent close policy

**Why?**
When the parent workflow closes, we need to apply the parent close policy on children that are under different domains and clusters.

**How did you test it?**
transferActiveTaskExecutorSuite

**Potential risks**

**Release notes**

**Documentation Changes**

### CODE REVIEW

File: common/persistence/dataManagerInterfaces.go [hunk 1]
```diff
@@ -610,6 +618,13 @@ type (
 		TargetCluster string
 	}
 
+	// CrossClusterApplyParentClosePolicyTask is the cross-cluster version of ApplyParentClosePolicyTask
+	CrossClusterApplyParentClosePolicyTask struct {
+		ApplyParentClosePolicyTask
```
Comment (demirkayaender) [inline common/persistence/dataManagerInterfaces.go:623]: We can actually flatten this struct and only use `CrossClusterApplyParentClosePolicyTask`; not sure if we wanted to keep the convention. Let me know what you think.

File: common/types/shared.go [hunk 2]
```diff
@@ -10696,13 +10696,70 @@ func (v *CrossClusterRecordChildWorkflowExecutionCompleteRequestAttributes) GetI
...
+func (v *AppyParentClosePolicyAttributes) GetParentClosePolicy() (o *ParentClosePolicy) {
+	if v != nil {
+		return v.ParentClosePolicy
+	}
+	return
+}
+
+type CrossClusterApplyParentClosePolicyRequestAttributes struct {
+	AppyParentClosePolicyAttributes []*AppyParentClosePolicyAttributes `json:"appyParentClosePolicyAttributes,omitempty"`
+	ParentClosePolicy               *ParentClosePolicy                 `json:"parentClosePolicy,omitempty"`
```
Comment (yycptt) [inline common/types/shared.go:10740]: we can remove this field I think?

File: service/history/execution/mutable_state_task_generator.go [hunk 3]
```diff
@@ -82,6 +82,10 @@ type (
 			transferTask *persistence.TransferTaskInfo,
 			targetCluster string,
 		) error
+		GenerateCrossClusterApplyParentClosePolicyTask(
```
Comment (yycptt) [inline service/history/execution/mutable_state_task_generator.go:85]: nit: probably add a TODO here saying ideally we should merge this method with `GenerateCrossClusterTaskFromTransferTask`?

**Review by yycptt — APPROVED** [review]
LGTM. Some minor comments.

**Pull request merged**